src/main/java/org/truffleruby/language/loader/FeatureLoader.java

```diff
@@ -50,10 +56,35 @@
     private TruffleObject sulongLoadLibraryFunction;
     private Map<String, String> nativeLibraryMap;
 
+    private NativeFunction getcwd;
+    private static final int PATH_MAX = 1024; // jnr-posix hard codes this value
+    private int cwdBufferSize = PATH_MAX;
+    private Pointer cwdBuffer;
+
     public FeatureLoader(RubyContext context) {
         this.context = context;
     }
 
+    public void initialize(NativePlatform nativePlatform, FinalizationService finalizationService) {
+        final TruffleNFIPlatform nfi = nativePlatform.getTruffleNFI();
+        final Object size_t_typedef = nativePlatform.getRubiniusConfiguration().get("rbx.platform.typedef.size_t");
+        final String size_t = nfi.toNFIType(StringOperations.getString((DynamicObject) size_t_typedef));
+
+        this.getcwd = nfi.getFunction("getcwd", 2, "(pointer," + size_t + "):pointer");
+        this.cwdBuffer = Pointer.malloc(cwdBufferSize);
+        this.cwdBuffer.enableAutorelease(finalizationService);
+    }
+
+    private String getWorkingDirectory() {
```

Comment:
This method isn't thread-safe. I'm not sure if this is a problem in practice, since `require` should be thread-safe. But it might be something to consider.

Reply:
You are right, the buffer needs to be thread-local.